src/codegen/compile_enhanced.ml

```diff
@@ -7061,7 +7102,9 @@ module Serialization = struct
           inc_data_size (compile_unboxed_const 12L) ^^ (* |id| + |page_count| = 8 + 4 *)
           get_x ^^ Region.vec_pages env ^^ size env blob)
       | Mut t ->
-        size_alias (fun () -> get_x ^^ MutBox.load_field env ^^ size env t)
+         size_alias (fun () -> get_x ^^ MutBox.load_field env ^^ size env t)
+      | Weak t ->
+         size_alias (fun () -> get_x ^^ WeakRef.load_field env ^^ size env t)
```

Comment:
Dito

Reply:
Hmm, so weak ref's are stable and we always generate deserialization code, even if we don't use it. Providing no support for deserialization in the Weak t case just crashes the compiler. So we need some support.